spring-cloud-dataflow-rest-resource/src/main/java/org/springframework/cloud/dataflow/rest/resource/DeploymentStatus.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.dataflow.rest.resource;
+
+import org.springframework.util.Assert;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+
+/**
+ * Provides a typed enumeration of deployment statuses for Streams.
+ *
+ * @author Gunnar Hillert
+ *
+ */
+@JsonFormat(shape = JsonFormat.Shape.OBJECT)
+public enum DeploymentStatus {
```

Comment:
The choice of name `DeploymentStatus` is dangerous IMHO, as the vocabulary is already very crowded with similar terms.

If you take into account my other comment, all this does is bridge a name to a human readable description. What's the value of bringing in an enum (on which you don't even do `==`?). Also, the code should make sure (eg in a static block, but without coupling this to `DeploymentState`) that all values of `DeploymentState` are covered (in case we change/add them in the future)